rack-protection/lib/rack/protection/base.rb

```diff
@@ -8,6 +8,21 @@
 
 module Rack
   module Protection
+    ##
+    # Base class for middlewares provided by rack-protection.
+    #
+    # == Options
+    #
+    # These may be given to any subclass, however they may affect different subclasses differently.
+    #
+    # [<tt>:reaction</tt>] - name of a method of the class that will be used to respond when the middleware rejects the response. Default behavior is to use #default_reaction.
```

Comment:
These options are all really useful and I didn't know about them until I started reading the source.